### ISSUE TITLE
cast Matrix to 2d array

### ISSUE THREAD

**Issue by kapsmane** [jmp75/rdotnet#30]
Firstly, it seems i cannot cast CharacterMatrix to string(,)

Second, NumericMatrix to Double(,) does not handle NA correctly:
suppose 
ma=matrix(0.1,3,3);ma[1,2]=NA
and then r is the matrix result:
                            Dim mym As NumericMatrix = r.AsNumericMatrix
                            Dim currentdfarray1 As Double(,) = mym.ToArray
                            somerange.Value = currentdfarray1

it writes to excel successfully, but ma[1,2] is shown as 65535 instead of empty

**Comment by kapsmane** [comment]
i'm sorry, the problem seems happen at .Net to Excel side, not r.net. plz close thx

**Comment by kapsmane** [comment]
i'm sorry, the second problem seems happen at .Net to Excel side, not r.net.

**Comment by kkkaaapppsss** [comment]
in bottom of CharacterMatrix.cs

 protected override string[,] GetArrayFast()
        {
            int rows = this.RowCount;
            int cols = this.ColumnCount;
            string [,] arr = new string[rows,cols];

            for (int i = 0; i < rows; i++)
            {
                for (int j = 0; j < cols; j++)
                {
                    arr[i, j] = this[i, j];
                }
            }
            return arr;
            //throw new NotSupportedException();
            //var values = new string[this.ItemCount];
            //Marshal.Copy(DataPointer, values, 0, values.Length);
            //return ArrayConverter.ArrayConvertAllTwoDim(values, this.RowCount, this.ColumnCount);
        }
basically i use the "slow" method and make it work. maybe we should include this in next release?
